### PR TITLE
evaluation refactor

### PR DESCRIPTION
**Pull request by VocodexElysium**
## ✨ Description

This pull request includes an overall refactor to the Amphion Evaluation module for faster inference speed and more computational modes for compatibility, which includes:
- **CUDA support:** Previously, all metrics are computed on the CPU. However, the computational speed will be incompatible with large-scale test sets that contain more than 10000 utterances. Currently, except for MCD and PESQ, which utilize totally encapsulated external packages, all the other metrics can now be computed on the GPU device to speed up.
- **Support for computing intelligibility-related metrics with the ground truth transcript:** Previously, CER and WER are computed on the transcript of an ASR model between the ground truth and the predicted audio. However, the ground truth transcript is available for tasks like TTS. Currently, it is possible to compute CER and WER between the ground truth transcript and the output of applying the ASR model on predicted audio.
- **Support for similarity computation without reference or ground truth:** Previously, speaker similarity is computed between a series of ground truth/prediction audios. However, such audio pairs may not be available for tasks like SVC. Currently, speaker similarity can be computed in a reference-free way that calculates the average score between all possible audio pairs in the reference and generated folder.

## 🚧 Related Issues

[110](https://github.com/open-mmlab/Amphion/issues/110)

## 👨‍💻 Changes Proposed

- ✅ CUDA support
- ✅ Intelligibility-related metrics refactored
- ✅ Similarity-related metrics refactored
- ✅ Algorithm speed optimization
- ✅ Bug fix

## 🧑‍🤝‍🧑 Who Can Review?

@lmxue @HeCheng0625 @zhizhengwu 

## ✅ Checklist

- ✅  Code has been reviewed
- ✅  Code complies with the project's code standards and best practices
- ✅  Code has passed all tests
- ✅  Code does not affect the normal use of existing features
- ✅  Code has been commented properly
- ✅  Documentation has been updated (if applicable)

### CODE REVIEW

**Review by VocodexElysium — COMMENTED** [review]
Able to merge

File: egs/metrics/README.md [hunk 1]
```diff
@@ -24,10 +24,10 @@ Until now, Amphion Evaluation has supported the following objective metrics:
...
   - Scale Invariant Signal to Noise Ratio (SISNR)
 - **Speaker Similarity**:
-  - Cosine similarity based on [Rawnet3](https://github.com/Jungjee/RawNet)
-  - Cosine similarity based on [Resemblyzer](https://github.com/resemble-ai/Resemblyzer)
-  - Cosine similarity based on [WavLM](https://github.com/microsoft/unilm/tree/master/wavlm)
-  - Cosine similarity based on [WeSpeaker](https://github.com/wenet-e2e/wespeaker) (👨‍💻 developing)
+  - Cosine similarity based on:
+    - [Rawnet3](https://github.com/Jungjee/RawNet)
+    - [Resemblyzer](https://github.com/resemble-ai/Resemblyzer)
+    - [WavLM](https://github.com/microsoft/unilm/tree/master/wavlm)
```
Comment (RMSnow) [inline egs/metrics/README.md:30]: Please modify the wavlm link to this one: https://huggingface.co/microsoft/wavlm-base-plus-sv

File: evaluation/metrics/similarity/speaker_similarity.py [hunk 2]
```diff
@@ -47,73 +48,131 @@ def extract_speaker_embd(
...
-    gpu = False
-    model.load_state_dict(
-        torch.load(
-            "pretrained/rawnet3/model.pt",
-            map_location=lambda storage, loc: storage,
-        )["model"]
-    )
-    model.eval()
-    print("RawNet3 initialised & weights loaded!")
+def extract_similarity(path_ref, path_deg, **kwargs):
```
Comment (RMSnow) [inline evaluation/metrics/similarity/speaker_similarity.py:51]: Add the comments for `overall` and `pairwith`.  You can list the typical scenario to exemplify (like [this](https://github.com/open-mmlab/Amphion/tree/main/egs/svc/MultipleContentsSVC#train-from-existing-source)). I'm a little confusing: do we need to choose `overall` when there are NOT ground-truth reference audios, and to choose `pairwith` when there are ground truths?

File: pretrained/wavlm/README.md [hunk 3]
```diff
@@ -0,0 +1,4 @@
+# Download
+
+- [Link](https://huggingface.co/microsoft/wavlm-base-sv/tree/main)
```
Comment (RMSnow) [inline pretrained/wavlm/README.md:3]: Same comments for URL.

**Review by RMSnow — CHANGES_REQUESTED** [review]
Thank you for such a hard work!

File: pretrained/wavlm/README.md (hunk 3 above)
Comment (VocodexElysium) [reply pretrained/wavlm/README.md]: fixed

File: pretrained/README.md [hunk 4]
```diff
@@ -212,3 +212,15 @@ Amphion
  ┃ ┃ ┣ tokenizer.json
  ┃ ┃ ┣ vocab.json
 ```
+
+## wavlm
+
+The official pretrained wavlm checkpoints can be available [here](https://huggingface.co/microsoft/wavlm-base-sv/tree/main). The file structure tree is as follows:
```
[the remark this answers is not in the document]
Comment (VocodexElysium) [reply pretrained/README.md]: we need wavlm-base-sv
I have changed all those URLs

File: egs/metrics/README.md (hunk 1 above)
Comment (VocodexElysium) [reply egs/metrics/README.md]: fixed

File: evaluation/metrics/similarity/speaker_similarity.py (hunk 2 above)
Comment (VocodexElysium) [reply evaluation/metrics/similarity/speaker_similarity.py:51]: Example added in the egs/metrics/README.md

**Comment by VocodexElysium** [comment]
> Please refine for WavLM features extraction. Support for both huggingface pretrained model and offline model.

support added

**Review by RMSnow — APPROVED** [review]

**Pull request merged**